### ISSUE TITLE
Black screen on InitializeLoader() and StopSubsystems() or app crash! (Unity)

### ISSUE THREAD

**Issue by iLyxa3D** [googlevr/cardboard#117]
Windows 10 / Unity 2020.1.6f / GoogleVR XR Cardboard 1.2.0
Device: Samsung Galaxy S7, Android

**Project**: 
- default GoogleVR sample for Unity

**Changes**: 
- turned off startup VR on app start
- start in Portrait, go LandscapeLeft before init vr loader
- init vr loader manual by ui button
- start subsystems by ui button
- stop subsystems by X button

**Problems**:
- after initloader we can see black screen and UI in overlay mode , 3D scene do not render (bug)
- after stop subsystems we can see black screen (with ui, but without scene objects), until deinitialize loader execute (bug)
- after [init loader - start subs - stop subs - deinit loader] we press init loader again and it lead to app crash! (critical)

APK demo: https://mega.nz/file/90plQByR#N3_WOaNVyG7tad4Rmerij1EfcmSD_D6ICs_42QPHFoM
Unity project: https://mega.nz/file/k1gXiBLA#bpR-z2UTxK90HdhDOQd1O4VBhmZHvWXAIpIDWr5fvFM

**Log**: 
```
09-22 11:59:53.978 10479 10496 D Unity   : [Subsystems] Loading plugin cardboard_api for subsystem Display...: com.ilyxa.testgooglexr
09-22 11:59:54.019 10479 10496 E CRASH   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***: com.ilyxa.testgooglexr
09-22 11:59:54.019 10479 10496 E CRASH   : Version '2020.1.6f1 (fc477ca6df10)', Build type 'Development', Scripting Backend 'il2cpp', CPU 'arm64-v8a': com.ilyxa.testgooglexr
09-22 11:59:54.019 10479 10496 E CRASH   : Build fingerprint: 'samsung/heroltexx/herolte:8.0.0/R16NW/G930FXXU8ETF2:user/release-keys': com.ilyxa.testgooglexr
09-22 11:59:54.019 10479 10496 E CRASH   : Revision: '8': com.ilyxa.testgooglexr
09-22 11:59:54.019 10479 10496 E CRASH   : ABI: 'arm64': com.ilyxa.testgooglexr
09-22 11:59:54.019 10479 10496 E CRASH   : Timestamp: 2020-09-22 11:59:54+0300: com.ilyxa.testgooglexr
09-22 11:59:54.019 10479 10496 E CRASH   : pid: 10479, tid: 10496, name: UnityMain  >>> com.ilyxa.testgooglexr <<<: com.ilyxa.testgooglexr
09-22 11:59:54.019 10479 10496 E CRASH   : uid: 10469: com.ilyxa.testgooglexr
09-22 11:59:54.019 10479 10496 E CRASH   : signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x10: com.ilyxa.testgooglexr
09-22 11:59:54.019 10479 10496 E CRASH   : Cause: null pointer dereference: com.ilyxa.testgooglexr
09-22 11:59:54.019 10479 10496 E CRASH   :     x0  0000000000000000  x1  0000007bd7492010  x2  fffffffffffffff8  x3  0000007bda351b00: com.ilyxa.testgooglexr
09-22 11:59:54.019 10479 10496 E CRASH   :     x4  0000007bda351b78  x5  0000000000000004  x6  0000000000000000  x7  0000000000000000: com.ilyxa.testgooglexr
09-22 11:59:54.019 10479 10496 E CRASH   :     x8  0000007bda353588  x9  0000007b5a7f2200  x10 0000000000000000  x11 0000000000000000: com.ilyxa.testgooglexr
09-22 11:59:54.019 10479 10496 E CRASH   :     x12 0000000000000000  x13 0000000000000000  x14 0000000000000000  x15 0000000000000002: com.ilyxa.testgooglexr
09-22 11:59:54.019 10479 10496 E CRASH   :     x16 0000007bdc3e71d8  x17 0000007bfddaeaf0  x18 000000000000000a  x19 0000007bd7492010: com.ilyxa.testgooglexr
09-22 11:59:54.019 10479 10496 E CRASH   :     x20 0000007bd6f7df58  x21 0000007bda353588  x22 0000007b5a7965bc  x23 0000007bdc4c2000: com.ilyxa.testgooglexr
09-22 11:59:54.019 10479 10496 E CRASH   :     x24 0000007bd6f7df20  x25 0000007bd874b078  x26 0000007bdb4ddb98  x27 0000000000000924: com.ilyxa.testgooglexr
09-22 11:59:54.020 10479 10496 E CRASH   :     x28 0000007bd57d72c0  x29 0000007bda351960: com.ilyxa.testgooglexr
09-22 11:59:54.020 10479 10496 E CRASH   :     sp  0000007bda351890  lr  0000007b5a7965fc  pc  0000007b5a796bc4: com.ilyxa.testgooglexr
09-22 11:59:54.020 10479 10496 E CRASH   : : com.ilyxa.testgooglexr
09-22 11:59:54.020 10479 10496 E CRASH   : backtrace:: com.ilyxa.testgooglexr
09-22 11:59:54.020 10479 10496 E CRASH   :       #00 pc 0000000000067bc4  /data/app/com.ilyxa.testgooglexr-ywGVob4dcb5zvDsns1zpzQ==/lib/arm64/libcardboard_api.so (BuildId: dd90749eb9a668980e1490134d06648632804b11): com.ilyxa.testgooglexr
09-22 11:59:54.020 10479 10496 E CRASH   :       #01 pc 00000000000675f8  /data/app/com.ilyxa.testgooglexr-ywGVob4dcb5zvDsns1zpzQ==/lib/arm64/libcardboard_api.so (BuildId: dd90749eb9a668980e1490134d06648632804b11): com.ilyxa.testgooglexr

```

**Main file code**:
```

using Google.XR.Cardboard;
using UnityEngine.XR.Management;
using UnityEngine;
using System.Collections;


/// <summary>
/// Initializes Cardboard XR Plugin.
/// </summary>
public class CardboardStartup : MonoBehaviour
{
    /// <summary>
    /// Start is called before the first frame update.
    /// </summary>
    public void Start()
    {
        // Configures the app to not shut down the screen and sets the brightness to maximum.
        // Brightness control is expected to work only in iOS, see:
        // https://docs.unity3d.com/ScriptReference/Screen-brightness.html.
        Screen.sleepTimeout = SleepTimeout.NeverSleep;
        Screen.brightness = 1.0f;

        // Checks if the device parameters are stored and scans them if not.
     /*  // I do not need scan QR at app start
        if (!Api.HasDeviceParams())
        {
            Api.ScanDeviceParams();
        }
     */
      
    }

    /// <summary>
    /// Update is called once per frame.
    /// </summary>
    public void Update()
    {
        if (Api.IsGearButtonPressed)
        {
            Api.ScanDeviceParams();
        }

        if (Api.HasNewDeviceParams())
        {
            Api.ReloadDeviceParams();
        }

        if (Api.IsCloseButtonPressed)
        {
            OffVR();
        }

       
    }

    public void Init()  //by button click
    {
        StartCoroutine(Initialize());
    }

    IEnumerator Initialize()
    {
        Screen.orientation = ScreenOrientation.LandscapeLeft;
        yield return new WaitForSeconds(2f); // we need delay because of wrong manager initialization screen settings if run init right after
       
       StartCoroutine(XRGeneralSettings.Instance.Manager.InitializeLoader()); //bug: lead to black screen, unlit we run StartSubsystem() 
                                                                              //bug: app crash at this point on second time on/off vr mode

       //XRGeneralSettings.Instance.Manager.InitializeLoaderSync();  //same shit

        while (XRGeneralSettings.Instance.Manager.isInitializationComplete == false)
            yield return new WaitForEndOfFrame();

        Debug.Log("Init finished");
    }

    public void StartVR()  //by button click
    {
        XRGeneralSettings.Instance.Manager.StartSubsystems();
    }

    void OffVR()
    {
        XRGeneralSettings.Instance.Manager.StopSubsystems();
        Screen.orientation = ScreenOrientation.Portrait;
        XRGeneralSettings.Instance.Manager.DeinitializeLoader();  //bug: witout this line it lead to black screen on StopSubsystems();
                                                                  
    }
}
```

Any suggestion?

**Comment by dletozeun** [comment]
Hello,

First you can replace this
``` 
StartCoroutine(XRGeneralSettings.Instance.Manager.InitializeLoader());
...
while (XRGeneralSettings.Instance.Manager.isInitializationComplete == false)
            yield return new WaitForEndOfFrame();
```

by this:
`yield return XRGeneralSettings.Instance.Manager.InitializeLoader();`

But it should not change anything.

What works for me is, in our tests and put in Coroutines:

Enable VR:

```
yield return XRGeneralSettings.Instance.Manager.InitializeLoader();
if (XRGeneralSettings.Instance.Manager.activeLoader != null)
{
    XRGeneralSettings.Instance.Manager.StartSubsystems();
}
else
{
    Debug.LogError("Initializing XR Failed");
}
```

Disable VR:
```
if (XRGeneralSettings.Instance.Manager.activeLoader != nul
{
    XRGeneralSettings.Instance.Manager.StopSubsystems();
}
XRGeneralSettings.Instance.Manager.DeinitializeLoader();
```

Not deinitializing the `XRManagerSettings` instance doesn't really stop VR and interfere with the Unity scene rendering at the moment but it should be the case I think. It is how the Cardboard Unity plugin is implemented.

As for the crash, I have found a quick fix: #124 and elaborated a bit there: https://github.com/googlevr/cardboard/issues/98#issuecomment-702274170

**Comment by jballoffet** [comment]
This is a duplicate issue, please take a look at #98.
